modules/tensor_mechanics/include/utils/RankTwoScalarTools.h

```diff
@@ -467,4 +470,83 @@ getQuantity(const RankTwoTensorTempl<T> & tensor,
                  scalarOptions().getRawNames());
   }
 }
+
+template <typename T>
+T
+getCylindricalComponent(const RankTwoTensorTempl<T> & tensor,
+                        const MooseEnum & scalar_type,
```

Comment:
My problem with this is that `MooseEnum`s are not strongly typed. Nothing would prevent a user to pass in a _scalar directional component_ `MooseEnum` into that function. As you are only querying index numbers, this will give a result. I would suggest using `enum class`es here. My preferred pattern is to never `getParam` a `MooseEnum` into a `MosseEnum` member variable, but rather use the `getParam("moose_enum_name").getEnum<EnumClassType>()` and thus have the parameter value in a strongly typed member variable. Then the parameter of this function would be a _specific_ enum class.